include/boost/geometry/algorithms/detail/overlay/get_turn_info.hpp

```diff
@@ -111,6 +111,60 @@ struct base_turn_handler
         both(ti, condition ? operation_union : operation_intersection);
     }
 
+
+#if ! defined(BOOST_GEOMETRY_USE_RESCALING)
+    template
+    <
+        typename UniqueSubRange1,
+        typename UniqueSubRange2
+    >
+    static inline int side_with_distance_measure(UniqueSubRange1 const& range_p,
+            UniqueSubRange2 const& range_q,
+            int range_index, int point_index)
+    {
+        if (range_index >= 1 && range_p.is_last_segment())
+        {
+            return 0;
+        }
+        if (point_index >= 2 && range_q.is_last_segment())
+        {
+            return 0;
+        }
+
+        typedef typename select_coordinate_type
+                <
+                    typename UniqueSubRange1::point_type,
+                    typename UniqueSubRange2::point_type
+                    >::type coordinate_type;
+
+        typedef detail::distance_measure<coordinate_type> dm_type;
+
+        dm_type const dm = get_distance_measure(range_p.at(range_index), range_p.at(range_index + 1), range_q.at(point_index));
```

Comment:
What are these ranges? I don't remember. Shouldn't the type of the index rather be some kind of unsigned integral type like `size_t` or ideally defined by the `UniqueSubRange` type as e.g. `size_type`?

Reply:
We pass small ranges to be able to know the next points (point after two intersecting segments).
I will look what we normally do, assuming indeed we can pass size_t

Reply:
So the size of the range is 1, 2 or 3. Ok, bigger type is not needed then.

Reply:
Indeed, OK, thanks